scripts/core/directives/FiletypeIconDirective.ts

```diff
@@ -11,10 +11,13 @@ export default angular.module('superdesk.core.directives.filetypeIcon', [])
     .directive('sdFiletypeIcon', ['gettextCatalog', (gettextCatalog) => ({
         scope: {item: '='},
         link: function(scope, element, attrs) {
-            var stopWatch = scope.$watch('item', (item) => {
-                if (item) {
+            var oldClass = null;
+
+            scope.$watch('item', (item) => {
+                if (item != null) {
                     initIcon(item);
```

Comment:
Rename to updateIcon

Reply:
Maybe setIcon since is not always an update

Reply:
That works too